### PR TITLE
runtime: Warn if `XDG_RUNTIME_DIR` is set but is not writable.

### PR DESCRIPTION
**Pull request by flouthoc**
If user is `rootless` and `XDG_RUNTIME_DIR` is found, podman will not proceed with `/tmp` directory it will try to use existing `XDG_RUNTIME_DIR` and if current user has no write access to `XDG_RUNTIME_DIR` podman session  will fail later

### CODE REVIEW

**Comment by flouthoc** [comment]
@giuseppe @Luap99 @mheon @rhatdan @TomSweeneyRedHat Moved resolution guide to troubleshooting page, could you all please take a look.

File: libpod/runtime.go [hunk 1]
```diff
@@ -328,6 +330,16 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (retErr error) {
 
 	runtime.mergeDBConfig(dbConfig)
 
+	unified, _ := cgroups.IsCgroup2UnifiedMode()
+	if unified && rootless.IsRootless() && !systemd.IsSystemdSessionValid(rootless.GetRootlessUID()) {
+		// If user is rootless and XDG_RUNTIME_DIR is found, podman will not proceed with /tmp directory
+		// it will try to use existing XDG_RUNTIME_DIR
+		// if current user has no write access to XDG_RUNTIME_DIR we will fail later
+		if unix.Access(runtime.storageConfig.RunRoot, unix.W_OK) != nil {
```
Comment (giuseppe) [inline libpod/runtime.go:338]: I am not sure this is the right place for the check.

`runtime.storageConfig.RunRoot` could have been set for a different reason than `XDG_RUNTIME_DIR` being specified.

Would it work if we have it in `SetXdgDirs` for the case where `os.Getenv("XDG_RUNTIME_DIR") != ""` ?

Comment (flouthoc) [reply libpod/runtime.go:338]: @giuseppe first version of PR had check exactly where you suggested, moved it after @mheon suggestion  reason is added in the comment itself https://github.com/containers/podman/pull/11327#issuecomment-906401764

Comment (mheon) [reply libpod/runtime.go:338]: This seems more sane - there's no guarantee Podman is actually using XDG_RUNTIME_DIR (we can fall back to other directories). This guarantees that we are checking whatever the currently-configured temporary files directory is.

Comment (giuseppe) [reply libpod/runtime.go:338]: thanks for the clarification.  It makes sense.

File: libpod/runtime.go [hunk 2]
```diff
@@ -328,6 +330,16 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (retErr error) {
...
+			logrus.Warnf("XDG_RUNTIME_DIR is pointing to a path which is not writable. Most likely podman will fail.")
```
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply libpod/runtime.go:339]: I'd use this here too, to be consistant.

**Comment by flouthoc** [comment]
assigning @mheon for review since he is already writing stuff on this.

**Comment by mheon** [comment]
LGTM

**Pull request merged**